draft-rha-jose-hpke-encrypt.md

```diff
@@ -220,44 +220,86 @@ This example demonstrates the representaton of an encapsulated key as a JWK.
 
 ### Integrated Encryption
 
-This mode only supports a single recipient and JWE Compact serialization.
+This mode only supports a single recipient.
 
 HPKE is employed to directly encrypt the plaintext, and the resulting ciphertext is included in the JWE ciphertext. 
 
-The sender MUST specify the 'epk' and 'alg' parameters in the protected header to indicate the use of HPKE. 
+In Integrated Encryption mode:
 
-The 'enc' (Encryption Algorithm) parameter MUST NOT be present. 
+*  The "epk" Header Parameter MUST be present, it MUST contain an Encapsulated JSON Web Key and it MUST occur only within the JWE Protected Header.
 
-This is a deviation from the rule in Section 4.1.2 of {{RFC7516}}. 
+*  The "alg" Header Parameter MUST be "HPKE-IntEnc", "enc" MUST be an HPKE algorithm from JSON Web Signature and Encryption Algorithms in {{JOSE-IANA}} and they MUST occur only within the JWE Protected Header.
 
-In Integrated Encryption mode:
+*  The JWE Ciphertext MUST be the base64url encoded 'ct' value.
+
+*  The JWE Initialization Vector value MUST be absent. 
+
+*  The JWE Authentication Tag MUST be absent.
+
+*  The JWE Encrypted Key MUST be absent.
+
+*  The HPKE "aad" parameter MUST be set to the JWE Additional Authenticated Data encryption parameter defined in Step 14 of Section 5.1 of {{RFC7516}} as input. 
 
-- The JWE Ciphertext MUST be the base64url encoded 'ct' value.
-- The JWE Initialization Vector value MUST be empty. 
-- The JWE Authentication Tag MUST be empty.
-- The JWE Encrypted Key MUST be empty.
-- The "aad" parameter MUST take the Additional Authenticated Data encryption parameter defined in Step 14 of Section 5.1 of {{RFC7516}} as input. 
 
 The following example demonstrates the use of Integrated Encryption with Compact Serialization:
 
 ~~~
-eyJhbGciOiJIUEtFLUJhc2UtUDI1Ni1TSEEyNTYtQUVTMTI4R0NNIiwiZXBrIjp7Imt0eSI6IkVLIiwiZWsiOiJCQU9TeWV3M05JLUkwNEd2WU1MT3Y0cDBEVG5WMWZjWnBFVW10dGs0YkRTdDAtakxzY0FDN3h3MjdORTFHZ0VuMUgtM3ZXSFA5eW1BOHl4aFRmVDBkYjQifX0...afBw3T1hUNjci4qq3ZZ-9KxnttB0iCEO_GUqbIStqYqB5DgRDpyYSuvoH1mMA31qKPqB41ld5mSP34yUys6WJM7nstDJ1-4nqUdhRpgfkGTECA.
+eyJhbGciOiJkaXIiLCJlbmMiOiJIUEtFLUJhc2UtUDI1Ni1TSEEyNTYtQUVTMTI4R0NNIiwiZXBrIjp7Imt0eSI6IkVLIiwiZWsiOiJCR05ranp0MDc2YnNSR2o3OGFYNUF6VF9IRU9KQmJZOXEyWm9fNWU3dGJLMGFQcXU0ZVQxV0kxNmp2UmxacXBNeXFaZlAtUndSNEp3dGF6XzhVOXRoWEEifX0...DG3qygxcMHw3iACy5mX_T4N4EqWc03W0nkTHjMJsC4nb6JS6vVj6wTGdlr5TOSr0ykaoyzpePXEvEyHhvpUwCyQQr6kbGlGuZsrJdUbZ728vmA.
+~~~
+{: #integrated-encryption-compact align="left" title="Integrated Encryption with Compact Serialization"}
+
+In the above example, the JWE Protected Header value is: 
+
+~~~
+{
+  "alg": ""HPKE-IntEnc",
```

Comment:
```suggestion
  "alg": "dir",
```

Reply:
I don't like this change, it introduces another algorithm that is conceptually very similar to "dir".

Reply:
It was discussed and agreed in https://github.com/tireddy2/JOSE_HPKE/pull/19.  Please have a look.

Reply:
It is somewhat calming that "dir" continues to mean an actual shared secret was used and I don't need to deduce from the "enc" value whether "anyone" could've produced this message or just the parties who know the symmetric secret.